src/libs/KeyboardShortcut/index.js

```diff
@@ -60,51 +50,12 @@ function getDisplayName(key, modifiers) {
     return displayName.join(' + ');
 }
 
-/**
- * Checks if an event for that key is configured and if so, runs it.
- * @param {Event} event
- * @private
- */
-function bindHandlerToKeydownEvent(event) {
-    if (!(event instanceof KeyboardEvent)) {
-        return;
-    }
-
-    const eventModifiers = getKeyEventModifiers(event);
-    const displayName = getDisplayName(event.key, eventModifiers);
-
-    // Loop over all the callbacks
-    _.every(eventHandlers[displayName], (callback) => {
-        // If configured to do so, prevent input text control to trigger this event
-        if (!callback.captureOnInputs && (
-            event.target.nodeName === 'INPUT'
-            || event.target.nodeName === 'TEXTAREA'
-            || event.target.contentEditable === 'true'
-        )) {
-            return true;
-        }
-
-        // Determine if the event should bubble before executing the callback (which may have side-effects)
-        let shouldBubble = callback.shouldBubble || false;
-        if (_.isFunction(callback.shouldBubble)) {
-            shouldBubble = callback.shouldBubble();
-        }
-
-        if (_.isFunction(callback.callback)) {
-            callback.callback(event);
-        }
-        if (callback.shouldPreventDefault) {
-            event.preventDefault();
-        }
-
-        // If the event should not bubble, short-circuit the loop
-        return shouldBubble;
-    });
-}
-
-// Make sure we don't add multiple listeners
-document.removeEventListener('keydown', bindHandlerToKeydownEvent, {capture: true});
-document.addEventListener('keydown', bindHandlerToKeydownEvent, {capture: true});
+_.each(CONST.KEYBOARD_SHORTCUTS, (shortcut) => {
+    KeyCommand.addListener(
+        shortcut.trigger[operatingSystem] || shortcut.trigger.DEFAULT,
+        (keycommandEvent, event) => bindHandlerToKeydownEvent(getDisplayName, eventHandlers, keycommandEvent, event),
+    );
+});
```

Comment:
We have 2 level of callback handlers:
- System level callback subscriptions
  - `document.addEventListener` for web
  - `keycommand.subscribe` for native
- Application level callbacks
  - `KeyboardShortcut.subscribe`
  
System level callbacks don't support attaching multiple listeners to the same topic (e.g. Ctrl + I), which is a requirement. This is used to have same subscription topic (e.g. Esc) to be used in multiple pages and add `priority` argument support.

The code above is used for the first level and is definitely aligned to existing pattern used in `KeyboardShortcut/index.js`.

Reply:
> We are attaching handlers for all the shortcuts

We are attaching a "system-level" shortcuts defined in CONST file.

> what if the user never subscribes to a key combination

Nothing happens, those are not attached at the application level hence won't affect React side.

> Can't call the KeyCommand.addListener( logic inside subscribe when the first callback is registered or first subscribe is called.

Can't. Web has `document.addEventListener` which is attached for every key event, where library gets only 1 per key-command. Overall this will complicate the code and doesn't bring any benefits. We have tried this pattern in https://github.com/Expensify/App/pull/14168.

Reply:
Now, right ?